### PR TITLE
cli: add env var for esm config

### PR DESCRIPTION
**Pull request by anshumanv**
Allows loading ESM config in CLI

- [x] Read and sign the [CLA][1]. PRs that haven't signed it won't be accepted.
- [x] Make sure your PR complies with the [writer's guide][2].
- [ ] Review the diff carefully as sometimes this can reveal issues.
- [ ] Do not abandon your Pull Request: [Stale Pull Requests][3].

### CODE REVIEW

**Review by rishabh3112 — DISMISSED** [review]

**Review by chenxsan — COMMENTED** [review]
I just notice there're actually two kinds of environment variables:

1. one is available to use in CLI, for example, `WEBPACK_CLI_SKIP_IMPORT_LOCAL=true npx webpack`
2. another is only available in `env` and the end users can't use it through CLI like the above kind, for example, it makes no sense to use `WEBPACK_BUILD=false npx webpack`.

Hence I would suggest separating them into two tables instead of one.

**Comment by anshumanv** [comment]
yeah makes sense to separate them

File: src/content/api/cli.md [hunk 1]
```diff
@@ -519,15 +519,21 @@ NODE_OPTIONS="--max-old-space-size=4096 -r /path/to/preload/file.js" webpack
...
-| `WEBPACK_PACKAGE`               | Use a custom webpack version in CLI                                 | no                 |
-| `WEBPACK_CLI_SKIP_IMPORT_LOCAL` | when `true` it will skip using the local instance of `webpack-cli`. | no                 |
+## CLI Environment Variables
+
+| Environment Variable                | Description                                                         |
+| ----------------------------------- | ------------------------------------------------------------------- |
+| `WEBPACK_CLI_SKIP_IMPORT_LOCAL`     | when `true` it will skip using the local instance of `webpack-cli`. |
+| `WEBPACK_CLI_FORCE_LOAD_ESM_CONFIG` | when `true` it will force load the ESM config.                      |
+| `WEBPACK_PACKAGE`                   | Use a custom webpack version in CLI.                                |
+
```
Comment (chenxsan) [inline src/content/api/cli.md:529]: Let's add an example here.

Comment (anshumanv) [reply src/content/api/cli.md:529]: done

File: src/content/api/cli.md [hunk 2]
```diff
@@ -527,6 +527,10 @@ NODE_OPTIONS="--max-old-space-size=4096 -r /path/to/preload/file.js" webpack
 | `WEBPACK_CLI_FORCE_LOAD_ESM_CONFIG` | when `true` it will force load the ESM config.                      |
 | `WEBPACK_PACKAGE`                   | Use a custom webpack version in CLI.                                |
 
+```bash
+WEBPACK_CLI_FORCE_LOAD_ESM_CONFIG=true webpack --config ./webpack.config.esm
```
Comment (chenxsan) [inline src/content/api/cli.md:531]:
```suggestion
WEBPACK_CLI_FORCE_LOAD_ESM_CONFIG=true npx webpack --config ./webpack.config.esm
```

Comment (anshumanv) [reply src/content/api/cli.md]: Done

**Review by chenxsan — APPROVED** [review]

**Pull request merged**